engine/player/unique_gear_shadowlands.cpp

```diff
@@ -1984,6 +1984,24 @@ void murmurs_in_the_dark( special_effect_t& effect )
   new dbc_proc_callback_t( effect.player, effect );
 }
 
+// id=336219 driver
+// id=336236 unknown use, triggered by driver
+// id=336234 damage spell
+// id=336222 melee damage spell?
+void dueling_form( special_effect_t& effect )
+{
+  struct dueling_form_t : public generic_proc_t
```

Comment:
I believe the standard practice (for shadowlands items at least) is to use `proc_spell_t` base here.

Reply:
Updated